modules/nf-core/bowtie/align/main.nf

```diff
@@ -3,9 +3,12 @@ process BOWTIE_ALIGN {
     label 'process_high'
 
     conda (params.enable_conda ? 'bioconda::bowtie=1.3.0 bioconda::samtools=1.15.1' : null)
+    def docker_image = params.docker_url != null ?
+        '${params.docker_url}/mulled-v2-ffbf83a6b0ab6ec567a336cf349b80637135bca3:676c5bcfe34af6097728fea60fb7ea83f94a4a5f-0' :
+        'quay.io/biocontainers/mulled-v2-ffbf83a6b0ab6ec567a336cf349b80637135bca3:676c5bcfe34af6097728fea60fb7ea83f94a4a5f-0'
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
         'https://depot.galaxyproject.org/singularity/mulled-v2-ffbf83a6b0ab6ec567a336cf349b80637135bca3:676c5bcfe34af6097728fea60fb7ea83f94a4a5f-0' :
-        'quay.io/biocontainers/mulled-v2-ffbf83a6b0ab6ec567a336cf349b80637135bca3:676c5bcfe34af6097728fea60fb7ea83f94a4a5f-0' }"
+        ${docker_url ?: 'quay.io/biocontainers'}/mulled-v2-ffbf83a6b0ab6ec567a336cf349b80637135bca3:676c5bcfe34af6097728fea60fb7ea83f94a4a5f-0' }"
```

Comment:
```nextflow
container { workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
    'https://depot.galaxyproject.org/singularity/mulled-v2-ffbf83a6b0ab6ec567a336cf349b80637135bca3:676c5bcfe34af6097728fea60fb7ea83f94a4a5f-0' :
    "${docker_url ?: 'quay.io/biocontainers'}/mulled-v2-ffbf83a6b0ab6ec567a336cf349b80637135bca3:676c5bcfe34af6097728fea60fb7ea83f94a4a5f-0' }
```

Personally, I think we should get away with using a closure. But nested GStrings should work too. 

@mahesh-panchal usually *knows* these things, whereas I am just guessing.

Reply:
This fails with 
```
    container { workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-ffbf83a6b0ab6ec567a336cf349b80637135bca3:676c5bcfe34af6097728fea60fb7ea83f94a4a5f-0' :
        "${docker_url ?: 'quay.io/biocontainers'}/mulled-v2-ffbf83a6b0ab6ec567a336cf349b80637135bca3:676c5bcfe34af6097728fea60fb7ea83f94a4a5f-0' }
```

Reply:
Configuration closures only work from the config. In the `process` block you need to have a GString leveraging the ternary expressions inside it. I really dislike lots of nested ternary expressions though as I think it makes the script look messy. With all these different registry options, I think they would be better off as profiles and the profile sets a `container_registry` parameter, which can then remove some of the existing logic since the image name and tag ( + build ) will same regardless of registry.